nextcord/command_client.py

```diff
@@ -0,0 +1,748 @@
+        self.type: Union[CommandType, CommandOptionType] = cmd_type
+        self.name: str = name
+        self.description: str = description
+        self.required: Optional[bool] = required
+        self.guild_ids: List[int] = guild_ids
+        self.choices: Dict[str, Any] = choices
+
+        self.cog_parent: Optional[CommandCog] = cog_parent
+        self.arguments: Dict[str, CommandArgument] = {}
+        self.children: Dict[str, ApplicationSubcommand] = {}
+        self._analyze_content()
+        self._analyze_callback()
+
+    def _analyze_content(self):
+        if self._parent and self._parent.type is CommandOptionType.sub_command_group and self.children:
+            raise NotImplementedError("A subcommand can't have both subcommand parents and children! Discord does not"
+                                      "support this.")
+        if isinstance(self._parent, ApplicationCommand):
+            if self.children:
+                self.type = CommandOptionType.sub_command_group
+            else:
+                self.type = CommandOptionType.sub_command
+        if self.type is CommandType.user or self.type is CommandType.message:
+            self.description = None
+        else:
+            if not self.description:
+                self.description = " "
+
+    def _analyze_callback(self):
+        if not self.name:
+            self.name = self._callback.__name__
+        first_arg = True
+
+        for value in signature(self.callback).parameters.values():
+            self_skip = value.name == "self"  # TODO: What kind of hardcoding is this, figure out a better way for self!
+            if first_arg:
+                # TODO: Is this even worth having?
+                if value.annotation is not value.empty and value.annotation is not Interaction:
+                    raise TypeError("First argument in an Application Command should be an Interaction.")
+                if self_skip:
+                    self_skip = False
+                else:
+                    first_arg = False
+            else:
+                arg = CommandArgument(value)
+                self.arguments[arg.name] = arg
+
+    @property
+    def callback(self) -> Callable:
+        return self._callback
+
+    async def call(self, state: ConnectionState, interaction: Interaction, option_data: List[Dict[str, Any]]):
+        """
+        Calls the callback, gathering and inserting kwargs into the callback as needed.
+        This must be able to call itself as subcommands may be subcommand groups, and thus have subcommands of their
+        own.
+
+        Parameters
+        ----------
+        state: :class:`ConnectionState`
+            State object used to fetch Guild, Channel, etc from cache.
+        interaction: :class:`Interaction`
+            Interaction object to pass to callback.
+        option_data: :class:`list`
+            List of option data dictionaries from interaction data payload.
+
+        Returns
+        -------
+
+        """
```

Comment:
```suggestion
            List of option data dictionaries from interaction data payload.
    """
```
it indeed does not return